File: chat-service/repository/entitty/document.py
from config.config import db
from sqlalchemy.dialects.postgresql import UUID, JSONB
from datetime import datetime
import uuid


class Document(db.Model):
    __tablename__ = "documents"
    __table_args__ = {"schema": "kb_project"}  # <-- thêm dòng này

    document_id = db.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    uploaded_by = db.Column(db.String(255), nullable=True)
    project_id = db.Column(UUID(as_uuid=True), db.ForeignKey("kb_project.projects.project_id"), nullable=False)
    folder_id = db.Column(UUID(as_uuid=True), db.ForeignKey("kb_project.folders.folder_id"), nullable=True)
    name = db.Column(db.String(255), nullable=True)
    file_path = db.Column(db.Text, nullable=True)
    file_type = db.Column(db.String(20), nullable=True)
    doc_metadata = db.Column("metadata", JSONB, nullable=True)
    uploaded_at = db.Column(db.DateTime, default=datetime.utcnow)

    def __repr__(self):
        return f"<Document {self.name} ({self.document_id})>"

File: project-service/src/main/java/com/review/projectservice/domain/entity/Document.java
package com.review.projectservice.domain.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.Map;
import java.util.UUID;

@Getter
@Setter
@Entity
@Data
@EntityListeners(AuditingEntityListener.class)
@Table(name = "documents", schema = "kb_project")
public class Document {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "document_id", nullable = false)
    private UUID id;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id", insertable = false, updatable = false)
    private Project project;

    @NotNull
    @Column(name = "project_id", nullable = false)
    private UUID projectId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "folder_id", insertable = false, updatable = false)
    private Folder folder;

    @Column(name = "folder_id")
    private UUID folderId;

    @Size(max = 255)
    @NotNull
    @Column(name = "name")
    private String name;

    @Column(name = "file_path", length = Integer.MAX_VALUE)
    private String filePath;

    @Size(max = 20)
    @Column(name = "file_type", length = 20)
    private String fileType;

    @Column(name = "is_active")
    private Boolean isActive;

    @Size(max = 50)
    @Column(name = "status", length = 50)
    private String status;


    @Column(name = "metadata")
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, Object> metadata;

    @CreatedBy
    @Column(name = "uploaded_by")
    private String uploadedBy;

    @CreationTimestamp
    @Column(name = "uploaded_at")
    private LocalDateTime uploadedAt;

}
